nnvm/tests/python/unittest/test_graph_gradient.py

```diff
@@ -130,7 +134,61 @@ def test_multi_loss_graph_gradients():
     # test reverse infer type for label
     assert grad_g.apply('InferType').json_attr('dtype_num_unknown_nodes') == 0
 
+def _run_op_grad(op):
```

Comment:
For operator unit test, we can add them under compiler/test_top_levelx.py, such as https://github.com/dmlc/tvm/blob/master/nnvm/tests/python/compiler/test_top_level1.py. Helper function supporting backward is implemented so that gradient can be tested here.